innkeeper/client.go

```diff
@@ -320,6 +318,43 @@ func getHttpError(r *http.Response) (error, bool) {
 	}
 }
 
+func (c *Client) writeRoute(url string, route *routeData) error {
+
+	res, err := json.Marshal(route)
+
+	req, err := http.NewRequest("POST", url, bytes.NewReader(res))
+
+	if err != nil {
+		return err
+	}
+
+	authToken, err := c.opts.Authentication.GetToken()
+
+	if err != nil {
+		return err
+	}
+
+	req.Header.Add(authHeaderName, authToken)
+	req.Header.Set("Content-Type", "application/json")
+	response, err := c.httpClient.Do(req)
+	if err != nil {
+		return err
+	}
+
+	defer response.Body.Close()
+
+	if response.StatusCode == http.StatusUnauthorized || response.StatusCode == http.StatusBadRequest {
```

Comment:
The problem with doing this is that in case of a 500, the calling code will believe that the request was successful, and can't even make a log entry or anything.